### ISSUE TITLE
Add support for maximum IPC request limit settable in IPC server

### ISSUE THREAD

**Issue by jfriesse** [ClusterLabs/libqb#76]
IRC backlog (I've forgot to fill right that day, sorry):

...
Jun 24 12:28:22 asalkeld -     is sends back the actual max size
Jun 24 12:28:41 asalkeld -     so if you ask for 40M
Jun 24 12:28:54 asalkeld -     and we only support 1M, then it returns 1M
Jun 24 12:29:07 honzaf  -      we?
Jun 24 12:29:10 asalkeld  -    and the client will use 1M
Jun 24 12:29:21 asalkeld -     we == server
Jun 24 12:29:37 asalkeld -     so just need some smarts in the server
Jun 24 12:29:45 asalkeld -     and ask for "lots"
Jun 24 12:29:50 asalkeld -     from the client
Jun 24 12:30:48 honzaf -       let's say I will call qb_ipcc_connect with 1Gig msg size
Jun 24 12:31:38 honzaf -       client will first create temp file (for shm /dev/shm) isn't it?
Jun 24 12:31:47 honzaf -       and then asks server
Jun 24 12:32:25 asalkeld -     no it uses sockets to begin with
Jun 24 12:33:08 honzaf  -      ok, and what is magic api call (called in server) to set max message size?
Jun 24 12:33:27 honzaf  -      and is there any INFIN number in qb_ipcc_connect?
Jun 24 12:34:05 asalkeld -     so no INF at the moment
Jun 24 12:34:32 asalkeld -     https://github.com/ClusterLabs/libqb/blob/master/lib/ipc_setup.c#L468
Jun 24 12:34:49 asalkeld -     maybe a special version of connection_accept?
Jun 24 12:35:07 asalkeld -     that takes the requested message size?
Jun 24 12:35:21 asalkeld -     (modifiable)
Jun 24 12:35:47 honzaf   -     yes and yes
Jun 24 12:37:07 honzaf   -     together with dispatch returning buffer (instead of caller prealocated one with fixed size)
and fix in totempg it would mean no longer 1Meg limit

In short. qb_ipcc_connect should take parameter like INFTIM and ipc server will be asked via special connection_accept method for maximum request (reply) size.

Of course if you are able to find out better method, I'm open to any suggestions. Idea is to have ONE limit in corosync (so ipc server) and ipc client will work with this limit (instead of having two limits, one for ipc server and one for ipc client)
